v2/slogger/rolling_file_appender/state.go

```diff
@@ -0,0 +1,64 @@
+package rolling_file_appender
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type state struct {
```

Comment:
Probably worth documenting this type is serialized/deserialized to/from disk. Changing this structure may cause failure. E.g: new fields won't be read from old files. Also, changing `LogStartTime` would also break reading an old file. It should probably explicitly name the field to write with a json tag/annotation.